tools/packaging/common/istio-iptables.sh

```diff
@@ -410,7 +410,7 @@ fi
 if [ -z "${DISABLE_REDIRECTION_ON_LOCAL_LOOPBACK-}" ]; then
   # Redirect app calls back to itself via Envoy when using the service VIP or endpoint
   # address, e.g. appN => Envoy (client) => Envoy (server) => appN.
-  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT
```

Comment:
Not sure if this is correct or fixes the problem. More comments in the PR description would help. Did you manually test that the loop is gone ? With or without split chains ? 

The IN filter has a 'original dst', so envoy will still connect to itself, and hit this rule in a loop.

Reply:
Sorry this PR is **NOT** the approach to fix the infinite loop we chatted yesterday. As stated in my comment...

This PR is suppose to 
- support the combination of split and inboundfilterchain to fix the bug only when filterchain is on
- not break anything when listener split is off (ISTIO_REREDIRECT and ISTIO_IN_REDIRECT is the same)